### DISCUSSION TITLE
How to use the command line tools?

### DISCUSSION THREAD

**Discussion by luweizheng** [apache/jena#1807] [Q&A] [answered]
Hi,
I download the source file. In the command line script, java needs to know the classpath of all the libraries as there is a setting`JENA_CP="$JENA_HOME"'/lib/*'`. I have already set the `JENA_HOME` variable. So what's the right way to get all the libs? I do not find any docs on how to get the libs.  Execute `mvn package` in the root folder? Or in the `apache-jena` folder?

**Comment by afs** [comment, marked as answer]
> the source file.

That would be `jena-4.7.0-source-release.zip`? That is the formal release artifact and is a buildable system.

The project provides binaries as well.

`apache-jena-4.7.0.zip` includes all the commands. Unpack the zip and the commands are in the `bin` and `bat` directories. No java compilation or coding required, just a Java runtime environment.

The `'/lib/*'` will set the classpath automatically (it's a feature of the Java JVM) and it is written to work in the filesystem layout of the binary `apache-jena-fuseki` distribution.

(zip file name corrected)

**Reply by luweizheng** [reply]
> > the source file.
> 
> That would be `jena-4.7.0-source-release.zip`? That is the formal release artifact and is a buildable system.
> 
> The project provides binaries as well.
> 
> `apache-jena-fuseki-4.7.0.zip` includes all the commands. Unpack the zip and the commands are in the `bin` and `bat` directories. No java compilation or coding required, just a Java runtime environment.
> 
> The `'/lib/*'` will set the classpath automatically (it's a feature of the Java JVM) and it is written to work in the filesystem layout of the binary `apache-jena-fuseki` distribution.

Yes, I download the `jena-4.7.0-source-release.zip` and unzip the file. 

`apache-jena-fuseki-4.7.0.zip` only contains fuseki. What I actually want to use is `riot`, `sparql`, and other command line tools. I find these command line tools in the `jena-4.7.0/apache-jena/bin`. Java needs to load the jars in the specified `lib` folder. Following the shell script defined in `jena-4.7.0/apache-jena/template.bin`, Java will go to `$JENA_HOME/lib/` to find the libraries. But there is currently no `lib` folder in the `jena-4.7.0-source-release.zip`. What do you think I should do? Thanks so much!

**Reply by kinow** [reply]
You will find the executable files mentioned before if you download the "Apache Jena Binary Distributions" :+1:

**Reply by luweizheng** [reply]
> You will find the executable files mentioned before if you download the "Apache Jena Binary Distributions" 👍

Thanks a lot!

**Reply by afs** [reply]
Sorry - my mistake. apache-jena-fuseki-4.7.0.zip should have been apache-jena-4.7.0.zip (fixed above)
